src/Common/Auth/AuthHash.php

```diff
@@ -195,14 +200,21 @@ public static function passwordVerify(&$password, $hash)
         } else {
             // Process algos supported by standard password_verify
             $valid = password_verify($password, $hash);
+
             if (!$valid) {
                 // Ensure do not need to process legacy hash (pre 5.0.0), which will get converted to standard hash
-                //  after a successful auth. The legacy hash has a malformed salt/hash combination which needs to
-                //  be adjusted and then run via crypt.
+                //  after a successful auth. This legacy hash was created with a salt of 21 characters rather than the standard
+                //  22 characters. Because of this, it does not work with above password_verify. Need to derive the salt
+                //  from the hash (up to the period character 29 in the hash). Note that this will not work on some
+                //  operating systems (for example, alpine linux crypt will return an error * instead of the hash because the
+                //  salt is not the correct length).
+                //
+                //  TODO: Consider removing this at some time in the near future since it overcomplicates authorization.
+                //
```

Comment:
hi @dcvidigal ,
Turns out that the fix will be hit or miss depending on the underlying operating system. The main crux of issue is OpenEMR was only using 21 characters for the salt prior to 5.0.0, which was one less the what is supposed to be used for blowfish (this was then fixed in OpenEMR 5.0.0). And crypt() support for short salts is operating system dependent. For example, on alpine linux, the crypt command returns the `*` error instead of the hash (this is why the automated testing checks whether the operating system's crypt() command will support the short salt). Will probably remove this block of code after a version or 2 (likely a year or so) to keep things simple in the future.